tests/testthat/test-get_vpts.R

```diff
@@ -0,0 +1,241 @@
+test_that("get_vpts() can fetch vpts data for a single radar and time", {
+  skip_if_offline()
+  single_radar_single_day <-
+    get_vpts(radar = "bejab", date = "2023-01-01", as_vpts = FALSE)
+  expect_s3_class(
+    # A known radar and date combination that ALOFT has data for
+    single_radar_single_day,
+    "data.frame"
+  )
+
+  # Expect a known length so no rows have been accidentally filtered away
+  expect_identical(
+    nrow(single_radar_single_day),
+    7125L
+  )
+})
+
+test_that("get_vpts() can fetch vpts data for multiple radars", {
+  skip_if_offline()
+  multiple_radars <- get_vpts(radar = c("bejab", "bewid"),
+                              date = "2023-01-01",
+                              as_vpts = FALSE)
+  expect_s3_class(
+    multiple_radars,
+    "data.frame"
+  )
+
+  expect_contains(
+    multiple_radars$radar,
+    c("bejab", "bewid")
+  )
+})
+
+test_that("get_vpts() can fetch data from multiple radar sources", {
+  skip_if_offline()
+  # Data from UVA
+  expect_contains(
+    get_vpts(radar = "bejab", date = "2018-02-02", as_vpts = FALSE)$source,
+    "uva"
+  )
+  # Data both on UVA and BALTRAD
+  expect_contains(
+    get_vpts(radar = "bejab", date = "2018-05-18", as_vpts = FALSE)$source,
+    c("uva", "baltrad")
+  )
+})
+
+test_that("get_vpts() returns columns of the expected type", {
+  skip_if_offline()
+
+  expected_col_types <-
+    list(
+      radar = "character",
+      source = "character",
+      datetime = c("POSIXct", "POSIXt"),
+      height = "integer",
+      u = "numeric",
+      v = "numeric",
+      w = "numeric",
+      ff = "numeric",
+      dd = "numeric",
+      sd_vvp = "numeric",
+      gap = "logical",
+      eta = "numeric",
+      dens = "numeric",
+      dbz = "numeric",
+      dbz_all = "numeric",
+      n = "integer",
+      n_dbz = "integer",
+      n_all = "integer",
+      n_dbz_all = "integer",
+      rcs = "numeric",
+      sd_vvp_threshold = "numeric",
+      vcp = "integer",
+      radar_latitude = "numeric",
+      radar_longitude = "numeric",
+      radar_height = "integer",
+      radar_wavelength = "numeric",
+      source_file = "character"
+    )
+
+  expect_identical(
+    get_vpts(radar = c("deflg"),
+             date = lubridate::ymd("20171015"),
+             as_vpts = FALSE) |>
+      purrr::map(class),
+    expected_col_types
+  )
+
+  # Specific radar causing trouble
+  expect_identical(
+    get_vpts(radar = c("dehnr"),
+             date = lubridate::ymd("20171015"),
+             as_vpts = FALSE) |>
+      purrr::map(class),
+    expected_col_types
+  )
+})
+
+test_that("get_vpts() can fetch data from a specific source only", {
+  skip_if_offline()
+
+  # Data from only BALTRAD even if UVA is available for the same interval
+  expect_identical(
+    get_vpts(radar = "bejab",
```

Comment:
Why does this return `BX42XXX` as the radar code?

Reply:
Probably because that's what's in the file on aloft.

Reply:
No, there is actually a bug here

Reply:
Added test for this bug f47046458ec48e4362cb73565c86520968fe8435

Keep in mind that the vpts files on aloftdata often have a different radar code encoded as in their filename. This seems to be correct. I suspect this is a UVA thing, for example: 

``` r
readr::read_csv("https://aloftdata.s3-eu-west-1.amazonaws.com/uva/daily/bejab/2014/bejab_vpts_20140202.csv")
#> Rows: 7200 Columns: 26
#> ── Column specification ────────────────────────────────────────────────────────
#> Delimiter: ","
#> chr   (2): radar, source_file
#> dbl  (21): height, u, v, w, ff, dd, sd_vvp, eta, dens, dbz, dbz_all, n, n_db...
#> lgl   (2): gap, vcp
#> dttm  (1): datetime
#> 
#> ℹ Use `spec()` to retrieve the full column specification for this data.
#> ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
#> # A tibble: 7,200 × 26
#>    radar datetime            height      u     v        w    ff    dd sd_vvp
#>    <chr> <dttm>               <dbl>  <dbl> <dbl>    <dbl> <dbl> <dbl>  <dbl>
#>  1 06410 2014-02-02 00:04:08      0 0.0647 1.33  -12.8     1.33  2.79  1.91 
#>  2 06410 2014-02-02 00:04:08    200 1.42   1.03  -11.0     1.76 54.2   2.73 
#>  3 06410 2014-02-02 00:04:08    400 1.20   0.206  -0.757   1.21 80.2   1.69 
#>  4 06410 2014-02-02 00:04:08    600 1.52   0.484  -0.114   1.59 72.3   1.52 
#>  5 06410 2014-02-02 00:04:08    800 1.37   0.387  -0.0908  1.42 74.2   1.50 
#>  6 06410 2014-02-02 00:04:08   1000 1.17   0.449  -0.312   1.25 69.0   1.35 
#>  7 06410 2014-02-02 00:04:08   1200 1.35   0.445   1.06    1.42 71.8   1.16 
#>  8 06410 2014-02-02 00:04:08   1400 1.15   0.457   1.25    1.24 68.3   1.22 
#>  9 06410 2014-02-02 00:04:08   1600 1.23   0.708   1.28    1.42 60.0   0.943
#> 10 06410 2014-02-02 00:04:08   1800 1.18   0.597   1.51    1.33 63.2   1.09 
#> # ℹ 7,190 more rows
#> # ℹ 17 more variables: gap <lgl>, eta <dbl>, dens <dbl>, dbz <dbl>,
#> #   dbz_all <dbl>, n <dbl>, n_dbz <dbl>, n_all <dbl>, n_dbz_all <dbl>,
#> #   rcs <dbl>, sd_vvp_threshold <dbl>, vcp <lgl>, radar_latitude <dbl>,
#> #   radar_longitude <dbl>, radar_height <dbl>, radar_wavelength <dbl>,
#> #   source_file <chr>
```

<sup>Created on 2024-11-18 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>